src/vs/editor/common/modes.ts

```diff
@@ -458,7 +470,7 @@ export interface SignatureHelpProvider {
 	/**
 	 * Provide help for the signature at the given position and document.
 	 */
-	provideSignatureHelp(model: model.ITextModel, position: Position, token: CancellationToken): ProviderResult<SignatureHelp>;
+	provideSignatureHelp(model: model.ITextModel, position: Position, token: CancellationToken, context: SignatureHelpContext): ProviderResult<SignatureHelp>;
 }
```

Comment:
Put token at the end where we can

Reply:
Agreed but we ran into this case with `CompletionItemProvider` too and decided it was better to put the `context` at the end rather than deal with the complexity of the function overload. We can discuss this more at the API sync